src/common/printing.c

```diff
@@ -163,3 +163,33 @@ void ffPrintUserString(const char* value)
         ++value;
     }
 }
+
+void ffPrintPercentBar(uint8_t percent, uint8_t green, uint8_t yellow, uint8_t red)
```

Comment:
Maybe add an assert / documentation that these numbers should add up to 100. I don't think this is intuitive.